### ISSUE TITLE
Skip certain MSBuild warnings when compile the cake script

### ISSUE THREAD

**Issue by mattleibow** [cake-build/cake#2668]
### What You Are Seeing?

When building a cake file that references NuGets that include dependencies that cake itself uses, I get a bunch of `CS1701` warnings:

```
warning CS1701: Assuming assembly reference 'Newtonsoft.Json, Version=11.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed' used by 'Cake.Json' matches identity 'Newtonsoft.Json, Version=12.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed' of 'Newtonsoft.Json', you may need to supply runtime policy
warning CS1701: Assuming assembly reference 'NuGet.Versioning, Version=4.6.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' used by 'Mono.ApiTools.NuGetDiff' matches identity 'NuGet.Versioning, Version=5.3.0.4, Culture=neutral, PublicKeyToken=31bf3856ad364e35' of 'NuGet.Versioning', you may need to supply runtime policy
...
```

### What is Expected?

A way to hide these warnings.

### What version of Cake are you using?

0.35.0

### Are you running on a 32 or 64 bit system?

64-bit

### What environment are you running on?  Windows? Linux? Mac?

Windows, Linux and Mac.

### Are you running on a CI Server?  If so, which one?

Local and DevOps.

### How Did You Get This To Happen? (Steps to Reproduce)

Just run a cake file with:

```
#addin nuget:?package=Cake.Json&version=4.0.0
#addin nuget:?package=Mono.ApiTools.NuGetDiff&version=1.3.0&loaddependencies=true
```

### Output Log

~~~sh
warning CS1701: Assuming assembly reference 'Newtonsoft.Json, Version=11.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed' used by 'Cake.Json' matches identity 'Newtonsoft.Json, Version=12.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed' of 'Newtonsoft.Json', you may need to supply runtime policy
warning CS1701: Assuming assembly reference 'NuGet.Versioning, Version=4.6.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' used by 'Mono.ApiTools.NuGetDiff' matches identity 'NuGet.Versioning, Version=5.3.0.4, Culture=neutral, PublicKeyToken=31bf3856ad364e35' of 'NuGet.Versioning', you may need to supply runtime policy
~~~

**Comment by augustoproiete** [comment]
Warnings are still present in Cake v1.0:

```
(3180,12): warning CS1701: Assuming assembly reference 'Newtonsoft.Json, Version=11.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed' used by 'Cake.Json' matches identity 'Newtonsoft.Json, Version=12.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed' of 'Newtonsoft.Json', you may need to supply runtime policy
(3187,12): warning CS1701: Assuming assembly reference 'Newtonsoft.Json, Version=11.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed' used by 'Cake.Json' matches identity 'Newtonsoft.Json, Version=12.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed' of 'Newtonsoft.Json', you may need to supply runtime policy
```

These warnings appear on a normal Cake run with default verbosity.

I suggest these warnings to be changed to `Diagnostic`,  which would still be helpful to troubleshoot when something goes wrong because an addin is not using the assembly that it expects, and at the same time removing the noise normal runs with default verbosity.

**Comment by augustoproiete** [comment]
This was resolved as part of #3338. Closing as dupe.
